### PR TITLE
Fix bug when deserializing a list of a non-map attribute

### PR DESCRIPTION
**Pull request by berdim99**

### CODE REVIEW

File: pynamodb/attributes.py [hunk 1]
```diff
@@ -523,7 +523,10 @@ def deserialize(self, values):
 
         lst_of_type = []
         for item in deserialized_lst:
-            lst_of_type.append(self.element_type(**item))
+            if issubclass(item.__class__, MapAttribute):
+                lst_of_type.append(self.element_type(**item))
+            else:
+                lst_of_type.append(item)
```
Comment (jmphilli) [inline pynamodb/attributes.py]: cc @danielhochman -- if the `of` type in the attribute passed in the constructor of the `ListAttribute` is not a subclass of `MapAttribute` then we don't want to explode the item (`**item` == explode)

**Comment by jmphilli** [comment]
i'm thinking we should block in the constructor of `ListATtribute` ... the `of` param should only take subclasses of `MapAttribute`... 

that said, we should check (test) that a list of unicode attributes works... i swear i made a test for that tho....

File: pynamodb/tests/test_attributes.py [hunk 2]
```diff
@@ -575,3 +575,33 @@ def test_map_of_list_of_map(self):
         }
         serialized = MapAttribute().serialize(family_attributes)
         self.assertDictEqual(MapAttribute().deserialize(serialized), family_attributes)
+
+    def test_list_of_map_with_of(self):
```
Comment (jmphilli) [inline pynamodb/tests/test_attributes.py]: can we add a test where a list of string / unicode attributes works?

**Comment by jmphilli** [comment]
lgtm @danielhochman ?

**Comment by berdim99** [comment]
@danielhochman - any thoughts on this?

**Comment by berdim99** [comment]
closing since Justin merged it all to Pynamo 2.0
